contentctl/actions/detection_testing/infrastructures/DetectionTestingInfrastructure.py

```diff
@@ -1401,6 +1482,7 @@ def hec_raw_replay(
     def status(self):
         pass
 
+    # TODO (cmcginley): the finish function doesn't actually stop execution
     def finish(self):
```

Comment:
It does do some extra cleanup, depending on whether it is running using containers or servers:

Removes the container when running using `containers`
https://github.com/splunk/contentctl/blob/98573b0fcc6040bc10046bf444c29d2685545ba6/contentctl/actions/detection_testing/infrastructures/DetectionTestingInfrastructureContainer.py#L29-L36


Nothing extra when running using `servers`: https://github.com/splunk/contentctl/blob/98573b0fcc6040bc10046bf444c29d2685545ba6/contentctl/actions/detection_testing/infrastructures/DetectionTestingInfrastructureServer.py#L10-L11

Reply:
Should this update the sync_obj.terminate field though?

Reply:
The _INTENT_ of the `.terminate` field is to indicate to all  running futures (whether they are doing container setup, execution, or shutdown) that they should stop early.   The `futures.as_completed` calls are SUPPOSED to catch any exceptions returned from the individual runners, for example:
https://github.com/splunk/contentctl/blob/8362af754acf327ec00daeb4bd5e189ae5722a84/contentctl/actions/detection_testing/DetectionTestingManager.py#L102
And set `.terminate = True` where appropriate.  So we shouldn't have to do this in the finish() function, or elsewhere, if an exception is raised.

But at the same time, I'm not extremely confident that we haven't missed some edge case where we shut down early, but DON'T raise an exception which is caught by the `future.result()` calls:
https://github.com/splunk/contentctl/blob/8362af754acf327ec00daeb4bd5e189ae5722a84/contentctl/actions/detection_testing/DetectionTestingManager.py#L100

Reply:
Gotcha. Yeah so that was exactly the issue that this PR fixes. Previously, errors were suppressed in setup, so even if setup failed, we would proceed to testing